src/query/functions-v2/src/scalars/string.rs

```diff
@@ -381,3 +442,90 @@ fn vectorize_string_to_string_2_arg(
         }
     }
 }
+
+fn register_bin(registry: &mut FunctionRegistry) {
+    registry.register_1_arg::<NumberType<u64>, StringType, _, _>(
+        "bin",
+        FunctionProperty::default(),
+        |_| None,
+        |val| format!("{:b}", val).as_bytes().to_vec(),
```

Comment:
you can directly `write!()` into writer.data, using register_passthrough_nullable_1_arg along with vectorize_with_writer_1_arg.